src/super_gradients/training/metrics/detection_metrics.py

```diff
@@ -45,17 +45,26 @@ def __init__(self, num_cls: int,
         super().__init__(dist_sync_on_step=dist_sync_on_step)
         self.num_cls = num_cls
         self.iou_thres = iou_thres
-        self.map_str = 'mAP@%.1f' % iou_thres[0] if not iou_thres.is_range() else 'mAP@%.2f:%.2f' % iou_thres
-        self.component_names = ["Precision", "Recall", self.map_str, "F1"]
+
+        if isinstance(iou_thres, IouThreshold):
+            self.iou_thresholds = iou_thres.to_tensor()
+        else:
+            self.iou_thresholds = torch.tensor([iou_thres])
+
+        self.map_str = 'mAP' + self._get_range_str()
+        self.component_names = [f"Precision{self._get_range_str()}",
+                                f"Recall{self._get_range_str()}",
+                                f"mAP{self._get_range_str()}",
+                                f"F1{self._get_range_str()}"]
+
         self.components = len(self.component_names)
         self.post_prediction_callback = post_prediction_callback
         self.is_distributed = super_gradients.is_distributed()
         self.denormalize_targets = not normalize_targets
         self.world_size = None
         self.rank = None
-        self.add_state("matching_info", default=[], dist_reduce_fx=None)
+        self.add_state(f"matching_info{self._get_range_str()}", default=[], dist_reduce_fx=None)
```

Comment:
Does this work?
self.add_state results in adding an attribute by that name. I doubt special characters like ":" will be legal here.
Anyways the mathicng_info is just internal for the metric and not a component name, so we can just drop the suffix self._get_range_str()} here.

Reply:
yes. in fact this is the only way this will work. otherwise, the collection of metrics tries to accumulate different IoU values into one metric. no problem with special chars, it is just a string